lib/internal/errors.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+
+// The whole point behind this internal module is to allow Node.js to no
+// longer be forced to treat every error message change as a semver-major
+// change. The NodeError classes here all expose a `code` property whose
+// value statically and permanently identifies the error. While the error
+// message may change, the code should not.
+
+const kCode = Symbol('code');
+const kSymbolsMap = new Map();
+
+function symbolFor(name) {
+  var sym = kSymbolsMap.get(name);
+  if (sym === undefined) {
+    sym = Symbol(name);
+    kSymbolsMap.set(name, sym);
+  }
+  return sym;
+}
+
+var assert, util;
+function lazyAssert() {
+  if (!assert)
+    assert = require('assert');
+  return assert;
+}
+
+function lazyUtil() {
+  if (!util)
+    util = require('util');
+  return util;
+}
+
+class NodeError extends Error {
+  constructor(key /**, ...args **/) {
+    lazyAssert().strictEqual(typeof key, 'string');
```

Comment:
This assert is not intended to be user facing. This is simply to make sure proper use of the internal API and is here only to make sure that node itself is passing in the right kind of value.